Cargo.toml

```diff
@@ -19,7 +19,8 @@ maintenance = { status = "experimental" }
 
 [dependencies]
 futures-task = "0.3"
```

Comment:
I wish we could get rid of this in favor of using `std::task::Wake`, which was stabilized in 1.51. Unfortunately, `std` doesn't contain an equivalent for `futures_task::waker_ref`, which would mean we either need to clone the `Arc` on every poll or keep two `Arc`s around in the wrapper all the time.

Reply:
I would like to avoid extra `fetch_add` at the hot path, although it doesn't violate a cache coherence most of the time.

Anyway, it's a subject of another PR.

Reply:
The "two Arc" solution would be to keep both the `Arc` we have right now which holds a concrete type that we can use to get at inner fields but cannot easily be used as a `Waker`, _and_ a `Waker::from(Arc::clone(&waker))` which can be passed in as the `Waker` to `poll` but doesn't have access to private fields.

But yes, for a different PR.